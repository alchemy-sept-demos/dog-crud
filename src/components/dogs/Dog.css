.dog-card {
  width: 300px;
  border-radius: 10px;
  border: 1px solid var(--color-blue);
  margin: 10px;
  text-align: center;
  box-shadow: #3d405b1f 0px 1px 3px, #3d405b3d 0px 1px 2px;
  cursor: pointer;
  height: 300px;
}

.dog-card img {
  width: 200px;
  border-radius: 50%;
}

.dog-detail {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  max-width: 500px;
}

.dog-detail img {
  width: 100%;
}

.dog-form {
  margin: 10px;
  display: flex;
  justify-content: center;
}

.controls {
  display: flex;
  flex-direction: column;
  align-items: center;
}
