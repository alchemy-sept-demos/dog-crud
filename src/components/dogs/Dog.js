import React from 'react';
import './Dog.css';

export default function Dog({ name, image, breed, bio, age, showDetail = false }) {
  const getClassName = () => {
    return showDetail ? 'dog-detail' : 'dog-card';
  };
  const getAgeString = () => {
    if (age === 0) {
      return 'less than one year old';
    } else {
      return `${age} year old`;
    }
  };
  return (
    <div className={getClassName()}>
      <h3>Meet {name}</h3>
      <img src={image} />
      <p>
        {name} is a(n) {getAgeString()} {breed}
      </p>
      {showDetail && (
        <div className="details">
          <p>{bio}</p>
        </div>
      )}
    </div>
  );
}
