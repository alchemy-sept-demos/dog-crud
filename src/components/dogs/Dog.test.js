import { render, screen } from '@testing-library/react';
import Dog from './Dog';

test('<Dog> should show the bio if showDetail=true', () => {
  // Render the component
  const { container } = render(
    <Dog
      name="Barton"
      image="https://placedog.net/500?id=6"
      age={1}
      breed="Weimaraner"
      bio="Cute dog"
      showDetail={true}
    />
  );
  // Wait for stuff to load
  const headerElement = screen.getByText('Meet Barton');
  expect(headerElement).toBeInTheDocument();

  const bioElement = screen.getByText('Cute dog');
  expect(bioElement).toBeInTheDocument();

  // Take a picture
  expect(container).toMatchSnapshot();
});
