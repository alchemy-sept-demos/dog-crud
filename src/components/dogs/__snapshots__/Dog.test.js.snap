// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dog> should show the bio if showDetail=true 1`] = `
<div>
  <div
    class="dog-detail"
  >
    <h3>
      Meet 
      Barton
    </h3>
    <img
      src="https://placedog.net/500?id=6"
    />
    <p>
      Barton
       is a(n) 
      1 year old
       
      Weimaraner
    </p>
    <div
      class="details"
    >
      <p>
        Cute dog
      </p>
    </div>
  </div>
</div>
`;
